dist/NoSleep.js

```diff
@@ -117,9 +117,9 @@ var oldIOS = function oldIOS() {
   return typeof navigator !== "undefined" && parseFloat(("" + (/CPU.*OS ([0-9_]{3,4})[0-9_]{0,1}|(CPU like).*AppleWebKit.*Mobile/i.exec(navigator.userAgent) || [0, ""])[1]).replace("undefined", "3_2").replace("_", ".").replace("_", "")) < 10 && !window.MSStream;
 };
 
-// Detect native Wake Lock API support
+// Detect native Wake Lock API support (Samsung Browser supports it but cannot use it)
 var nativeWakeLock = function nativeWakeLock() {
-  return "wakeLock" in navigator;
+  return "wakeLock" in navigator && window.navigator.userAgent.indexOf("Samsung") === -1;
```

Comment:
Isn't this not safe? On new Samsungs it might work. Is it possible to execute it, and in case the promise rejects, then we should be like "ok this samsung doesn't allow it, so then fallback"?